### PR TITLE
Improve the package overview doc

### PR DESCRIPTION
**Pull request by BuddhiWathsala**
## Purpose
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/925

## Approach
- Add the new file API information to the doc
- Remove unnecessary doc comments

### CODE REVIEW

File: io-ballerina/Package.md [hunk 1]
```diff
@@ -1,31 +1,68 @@
...
-This module is designed to support input and output operations via channels in a canonical way, either in a blocking,
-or non-blocking manner.
+The following diagram depicts the overview architecture of the I/O package.
 
 ![architecture](./resources/package-architecture.svg)
 
-### Channels
-A channel represents an I/O source or sink of some bytes, characters, or records that are opened for reading or
-writing respectively.
+The I/O package allows users to read from the console or a file.
```
Comment (praneesha) [inline io-ballerina/Package.md:7]:
```suggestion
The I/O package allows you to read from the console or a file. 
```

File: io-ballerina/Package.md [hunk 2]
```diff
@@ -1,31 +1,68 @@
...
+- JSON I/O
+- XML I/O
 
-### Byte channels
-The most primitive channel is the `ByteChannel` which reads and writes 8-bit bytes. For an example on the `ByteChannel`, see the [Byte I/O Example](https://ballerina.io/swan-lake/learn/by-example/byte-io.html).
 
-### Character channels
-The `CharacterChannel` is used to read and write characters. The charset encoding is specified when creating the `CharacterChannel`. For an example on the `CharacterChannel`, see the [Read/Write Files Example](https://ballerina.io/swan-lake/learn/by-example/character-io.html).
+### Console I/O
+The console I/O APIs are as follows, which helps the user to read from the console as well as write to the console.
```
Comment (praneesha) [inline io-ballerina/Package.md:17]:
```suggestion
The console I/O APIs, which help the user to read from the console as well as write to the console are as follows.
```

File: io-ballerina/Package.md [hunk 3]
```diff
@@ -1,31 +1,68 @@
...
-the respective type. For examples on reading/writing JSON or XML sources, see the [JSON I/O Example](https://ballerina.io/swan-lake/learn/by-example/json-io.html) and [XML I/O Example](https://ballerina.io/swan-lake/learn/by-example/xml-io.html).
+### Bytes I/O
+The bytes I/O APIs provide the reading and writing APIs in both streaming and non-streaming ways. Those APIs are,
+- `io:fileReadBytes`
+- `io:fileReadBlocksAsStream`
+- `io:fileWriteBytes`
+- `io:fileWriteBlocksFromStream`
 
-### Record channels
+Refer [this example](https://ballerina.io/swan-lake/learn/by-example/byte-io.html) to learn how to use bytes read and write APIs.
```
Comment (praneesha) [inline io-ballerina/Package.md:30]:
```suggestion
To learn how to use bytes read and write APIs, see the [Byte IO example](https://ballerina.io/learn/by-example/byte-io.html).
```

File: io-ballerina/Package.md [hunk 4]
```diff
@@ -1,31 +1,68 @@
...
 
-Also, Ballerina supports I/O for delimited records. For an example on reading the records in a text file, see the [Record I/O Example](https://ballerina.io/swan-lake/learn/by-example/record-io.html).
+### Strings I/O
+The strings I/O APIs provide the reading and writing APIs in 3 different ways:
+1. read the whole file content as a string and write a given string to a file
```
Comment (praneesha) [inline io-ballerina/Package.md:34]:
```suggestion
1. Read the whole file content as a string and write a given string to a file
```

File: io-ballerina/Package.md [hunk 5]
```diff
@@ -1,31 +1,68 @@
...
+1. read the whole file content as a set of lines and write a given set of lines to a file
```
Comment (praneesha) [inline io-ballerina/Package.md:35]:
```suggestion
1. Read the whole file content as a set of lines and write a given set of lines to a file
```

File: io-ballerina/Package.md [hunk 6]
```diff
@@ -1,31 +1,68 @@
...
+1. read the whole file content as a stream of lines and write a given stream of lines to a file
```
Comment (praneesha) [inline io-ballerina/Package.md:36]:
```suggestion
1. Read the whole file content as a stream of lines and write a given stream of lines to a file
```

File: io-ballerina/Package.md [hunk 7]
```diff
@@ -1,31 +1,68 @@
...
+The strings I/O APIs are as follows:
+- `io:fileReadString`
+- `io:fileReadLines`
+- `io:fileReadLinesAsStream`
+- `io:fileWriteLines`
+- `io:fileWriteLinesFromStream`
 
-### Data Channels
-Ballerina supports performing data i/o operations.
+Refer [this example](https://ballerina.io/swan-lake/learn/by-example/strings-io.html) to learn how to use bytes read and write APIs.
```
Comment (praneesha) [inline io-ballerina/Package.md:45]:
```suggestion
To learn how to use bytes read and write APIs, see the [Strings IO example](https://ballerina.io/learn/by-example/strings-io.html).
```

File: io-ballerina/Package.md [hunk 8]
```diff
@@ -1,31 +1,68 @@
...
 
-A Person object could be serialized into a file or a network socket as seen in the [Data I/O Example](https://ballerina.io/swan-lake/learn/by-example/data-io.html).
+### CSV I/O
+The CSV I/O APIs provide the reading and writing APIs in both streaming and non-streaming ways. Those APIs are,
+- `io:fileReadCsv`
+- `io:fileReadCsvAsStream`
+- `io:fileWriteCsv`
+- `io:fileWriteCsvFromStream`
 
+Refer [this example](https://ballerina.io/swan-lake/learn/by-example/csv-io.html) to learn how to use CSV read and write APIs.
```
Comment (praneesha) [inline io-ballerina/Package.md:54]:
```suggestion
To learn how to use CSV read and write APIs, see the [CSV IO example](https://ballerina.io/learn/by-example/csv-io.html).
```

File: io-ballerina/Package.md [hunk 9]
```diff
@@ -1,31 +1,68 @@
...
+
+### JSON I/O
+The JSON I/O APIs provide the reading and writing APIs for JSON content. Those APIs are,
+- `io:fileReadJson`
+- `io:fileWriteJson`
+
+Refer [this example](https://ballerina.io/swan-lake/learn/by-example/json-io.html) to learn how to use JSON read and write APIs.
```
Comment (praneesha) [inline io-ballerina/Package.md:61]:
```suggestion
To learn how to use JSON read and write APIs, see the [JSON IO example](https://ballerina.io/learn/by-example/json-io.html).
```

File: io-ballerina/Package.md [hunk 10]
```diff
@@ -1,31 +1,68 @@
...
+
+### XML I/O
+The XML I/O APIs provide the reading and writing APIs for XML content. Those APIs are,
+- `io:fileReadXml`
+- `io:fileWriteXml`
+
+Refer [this example](https://ballerina.io/swan-lake/learn/by-example/json-io.html) to learn how to use XML read and write APIs.
```
Comment (praneesha) [inline io-ballerina/Package.md:68]:
```suggestion
To learn how to use XML read and write APIs, see the [JSON IO example](https://ballerina.io/learn/by-example/json-io.html).
```

**Comment by BuddhiWathsala** [comment]
Suggestions are fixed in here https://github.com/ballerina-platform/module-ballerina-io/pull/72/commits/4289b5030b6977ea741410918dd969cab54f777e

**Review by praneesha — APPROVED** [review]

**Pull request merged**